test/e2e/actions/cloud/azure.go

```diff
@@ -392,6 +399,62 @@ func (a *AzureAction) deletePrivateEndpoint(ctx context.Context, endpointName st
 	return nil
 }
 
+func (a *AzureAction) CreateKeyVault(keyName string) (string, error) {
+	a.t.Helper()
+
+	ctx := context.Background()
+
+	client, err := azkeys.NewClient(AzureKeyVaultURI, a.credentials, nil)
+	if err != nil {
+		return "", err
+	}
+
+	params := azkeys.CreateKeyParameters{
+		KeySize: toptr.MakePtr[int32](2048),
+		Kty:     toptr.MakePtr(azkeys.KeyTypeRSA),
+	}
+
+	r, err := client.CreateKey(ctx, keyName, params, nil)
+	if err != nil {
+		return "", err
+	}
+
+	a.t.Cleanup(func() {
+		err = a.deleteKeyVault(ctx, keyName)
+		if err != nil {
+			a.t.Error(err)
+		}
+	})
+
+	keyID := string(*r.KeyBundle.Key.KID)
+
+	return keyID, nil
+}
+
+func (a *AzureAction) deleteKeyVault(ctx context.Context, keyName string) error {
+	a.t.Helper()
+
+	client, err := azkeys.NewClient(AzureKeyVaultURI, a.credentials, nil)
```

Comment:
As I imagined, this is not the latest SDK API and it differs from the usage in other methods.
See the package `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault`
You will be able to create a factory and use the `PollUntilDone()` from the returned operation.

Reply:
`armkeyvault` has [no functionality to delete keys](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault#KeysClient) (just create and get), as far as I can tell. I'm not sure why there is an additional module for keyvault like this, but it has delete functionality. Alternatively, I could use `armkeyvault` to create a new vault every time, which you _can_ delete and poll, but that might add some more work for figuring out permissions when creating the vault.